lib/src/lints/reverse_if_to_reduce_nesting/visitors/reverse_if_to_reduce_nesting_visitor.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:solid_lints/src/lints/reverse_if_to_reduce_nesting/visitors/if_statement_visitor.dart';
+
+/// The AST visitor that will collect all unnecessary if statements
+class ReverseIfToReduceNestingVisitor extends RecursiveAstVisitor<void> {
+  final _nodes = <AstNode>[];
+
+  /// All unnecessary if statements and conditional expressions.
+  Iterable<AstNode> get nodes => _nodes;
+
+  @override
+  void visitBlockFunctionBody(BlockFunctionBody node) {
+    super.visitBlockFunctionBody(node);
+    if (node.block.statements.isEmpty) return;
+
+    final first = node.block.statements.first;
+
+    switch (node.block.statements.length) {
+      case 1:
+        if (first is IfStatement) _handleIfStatement(first);
+        return;
+      case 2:
+        final last = node.block.statements.last;
+        if (last is ReturnStatement && first is IfStatement) {
+          _handleIfStatement(first);
+        }
+      case _:
+        return;
+    }
+  }
+
+  void _handleIfStatement(IfStatement node) {
+    if (_isElseIfStatement(node)) return;
+    if (_hasElseStatement(node)) return;
+    if (!_containsIfStatement(node.thenStatement)) return;
```

Comment:
Could you explain why this is needed?
```suggestion
    final containsNestedIfStatement = _containsIfStatement(node.thenStatement);
    if (!containsNestedIfStatement) return;
```